### PR TITLE
FEATURE: added display progress for file transferred using File Browser.

### PR DESCRIPTION
**Pull request by badescuga**
This handles zips & also multiple files added. Plan to add another drop down for viewing completed/in progress items being transferred.
screenshot_1

### CODE REVIEW

**Comment by davidebbo** [comment]
Played with it on test site and it seems to worked well! Haven't looked at code yet.

Only potential question is whether the extra work that it does might affect perf, though that's unlikely.

File: Kudu.Services.Web/Content/Scripts/FileBrowser.js [hunk 1]
```diff
@@ -41,7 +41,64 @@ var statusbar = {
         }
 }
 
+var copyProgressObjects = {};
+var copyObjectsManager = {
+    init: function() {
+        copyProgressObjects = {};
+    },
+    addCopyStats: function (uri, loadedData, totalData
+   )  {
```
Comment (suwatch01) [inline Kudu.Services.Web/Content/Scripts/FileBrowser.js]: nit: should this be with line above.

**Comment by snobu** [comment]
I can do the benchmark vanilla vs @badescuga 's progress-indicator version so there's no bias :)

File: Kudu.Services.Web/Content/Scripts/FileBrowser.js (hunk 1 above)
Comment (badescuga) [inline Kudu.Services.Web/Content/Scripts/FileBrowser.js]: do you mean that the code should be: 
`addCopyStats: function (uri, loadedData, totalData)  {` 

i.e. having `) {` on the line above?

**Comment by snobu** [comment]
Here comes the benchmark.

image

In the __Left__ corner: __Progress__ Indicator build
in the __Right__ corner: __Vanilla__ Kudu, as a new fresh Web App

Both sites in __West Europe__ in __Standard S1__.
Performance looks unaffected.

bench

File: Kudu.Services.Web/Content/Scripts/FileBrowser.js (hunk 1 above)
Comment (suwatch01) [inline Kudu.Services.Web/Content/Scripts/FileBrowser.js]: yes.

Comment (badescuga) [inline Kudu.Services.Web/Content/Scripts/FileBrowser.js]: I'm going to fix that in the next pull request, sometime in the next few days. I'm working on the manager for uploads. 


If everything else is ok, is it approvable in this condition?





On Fri, Oct 2, 2015 at 10:17 AM -0700, "suwatch01" <notifications@github.com> wrote:












In Kudu.Services.Web/Content/Scripts/FileBrowser.js:
> @@ -41,7 +41,64 @@ var statusbar = {
>          }
>  }
>  
> +var copyProgressObjects = {};
> +var copyObjectsManager = {
> +    init: function() {
> +        copyProgressObjects = {};
> +    },
> +    addCopyStats: function (uri, loadedData, totalData
> +   )  { 



yes.

**Comment by davidebbo** [comment]
It's in, thanks! Rebase and squashed as ab40cda49b1f696db34e7ad21b1d250be133314f.

**Pull request closed without merging**